### PR TITLE
Qt 5.11.1

### LINKED ISSUE #2830: Update to Qt 5.11
Qt 5.11 was released today.

http://blog.qt.io/blog/2018/05/22/qt-5-11-released/

There are a number of issues in 5.10.1 (our current release) which we know or suspect will be fixed in 5.11. (5.12 is the next LTS release, but won't come out until November.)

Also useful:

>  we now support embedded DevTools without requiring the use of a separate browser

### PR DESCRIPTION
**Pull request by gtritchie**
Moving RStudio desktop to Qt 5.11.1. This was a non-trivial update, especially on Windows.

- switch to MSVC 2017 on Windows
- switched to 64-bit only on Windows (this was done a few weeks ago, but was motivated by the switch to Qt 5.11.1, see [commit](https://github.com/rstudio/rstudio/commit/3d76fd0bb4b4d40f6fea6f5cc57e5122c691751b))
- switch to cmake 3.1.0 or above
- trusty and opensuse docker builds stay on Qt 5.10.1 due to linker issues: #3120, #3121 track continued investigation of doing custom Qt builds for those platforms (I ran into issues get QtWebEngine to build on those platforms; work in progress on that can be [seen here](https://github.com/gtritchie/qt-gst10/blob/master/Dockerfile.trusty-x86_64-qt11) but QtWebEngine fails to build)
    -  this means any code that depends on Qt 5.11 needs to be conditionally compiled so it still builds against 5.10; you can do this via `QT_VERSION_CHECK` macro
- new xenial dockerfile
- remove Linux workaround for Qt 5.10.1 bug that prevented finding QtWebEngineProcess if sandboxing enabled (leave it in place if compiling with 5.10.1)
- install Qt via online installer in dockerfiles and dependency scripts; takes longer and is larger, but makes dev and build environments more consistent
- option to use packaged download of Qt (way we used to do it)
- work in progress batch file for running Windows docker builds (works but requires juggling, described in batch file comments)
- created separate install-qt batch file for Windows, instead of doing it directly in install-dependencies batch file

Fixes #2830 

For sanity testing I ran container builds and installed the resulting packages on clean VMs (list below) and verified I could launch RStudio. Didn't test much beyond ability to launch.

| Dockerfile Used to Build      | Platform(s) Installed On |
| --------------------------- | ------------------------ |
| Dockerfile.windows              | Windows 10                     |
| Dockerfile.trusty-amd64     | Ubuntu 14.04.5               |
| Dockerfile.centos7-x86_64 | CentOS 7                         |
| Dockerfile.debian9-x86_64 | Debian 9                          |
| Dockerfile.xenial-amd_64    | Ubuntu 16.04.4               |
| Dockerfile.opensuse             | OpenSUSE 42.3             |
| Dockerfile.opensuse15         | OpenSUSE Leap (15)     |

I did **not** confirm that Windows build runs on **Windows 7 or 8**. That needs to be checked.

For Mac I built the package directly via make-package and installed the results on a different Mac.

I also tested bootstrapping a clean Windows-10 dev machine for RStudio/Qt5.11.1 development.

### CODE REVIEW

File: docker/win-docker-compile.cmd [hunk 1]
```diff
@@ -0,0 +1,102 @@
+@echo off
+setlocal enableextensions enabledelayedexpansion
+
+REM Build Windows docker image containing tools to build RStudio IDE
```
Comment (jmcphers) [inline docker/win-docker-compile.cmd]: This is very cool!

File: docker/jenkins/Dockerfile.centos7-x86_64 [hunk 2]
```diff
@@ -63,7 +62,8 @@ RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-common
 # install Qt SDK
 COPY dependencies/linux/install-qt-sdk /tmp/
 RUN mkdir -p /opt/RStudio-QtSDK && \
-    export QT_SDK_DIR=/opt/RStudio-QtSDK/Qt5.10.1 && \
+    export QT_SDK_DIR=/opt/RStudio-QtSDK/Qt5.11.1 && \
+    export QT_QPA_PLATFORM=minimal && \
```
Comment (kevinushey) [inline docker/jenkins/Dockerfile.centos7-x86_64]: Just out of curiosity, why is this necessary? (Hopefully it doesn't affect any of the platform-specific stuff that Qt does, e.g. choosing the appropriate theme for system file dialogs?)

I think that stuff normally happens at runtime but just want to double check that this change doesn't affect that.

Comment (gtritchie) [reply docker/jenkins/Dockerfile.centos7-x86_64]: `QT_QPA_PLATFORM` is purely for install-time; this allows silently running the online-installer without requiring the containers to have any form of X-Windows installed.

Comment (kevinushey) [reply docker/jenkins/Dockerfile.centos7-x86_64]: Got it! Thanks for the clarification.

Comment (kevinushey) [reply docker/jenkins/Dockerfile.centos7-x86_64]: Got it! Thanks for the clarification.
